packages/@webex/internal-plugin-metrics/src/ca-metrics.ts

```diff
@@ -0,0 +1,312 @@
+/* eslint-disable class-methods-use-this */
+/* eslint-disable valid-jsdoc */
+import {getOSNameInternal} from '@webex/internal-plugin-metrics';
+import {BrowserDetection} from '@webex/common/src/browser-detection';
+import uuid from 'uuid';
+import {merge} from 'lodash';
+import {anonymizeIPAddress, clearEmpty, userAgentToString} from './ca-metrics.util';
+import {CLIENT_NAME} from './config';
+import {Event as RawEvent} from './Event';
+import {FeatureEvent} from './FeatureEvent';
+import {ClientEvent} from './ClientEvent';
+import {CallAnalyzerLatencies} from './ca-metrics-latencies';
+
+type Event = Omit<RawEvent, 'event'> & {event: ClientEvent | FeatureEvent};
+
+const {getOSVersion, getBrowserName, getBrowserVersion} = BrowserDetection();
+
+type RecursivePartial<T> = {
+  [P in keyof T]?: T[P] extends (infer U)[]
+    ? RecursivePartial<U>[]
+    : T[P] extends object
+    ? RecursivePartial<T[P]>
+    : T[P];
+};
+
+type GetOriginOptions = {
+  clientType: NonNullable<Event['origin']['clientInfo']>['clientType'];
+  subClientType: NonNullable<Event['origin']['clientInfo']>['subClientType'];
+};
+
+type GetIdentifiersOptions = {
+  meeting?: any;
+  meetingLookupUrl?: string;
+  trackingId?: string;
+  mediaConnections?: any[];
+};
+
+/**
+ * @description Util class to handle Call Analyzer Metrics
+ * @export
+ * @class CallAnalyzerMetrics
+ */
+export class CallAnalyzerMetrics {
+  // eslint-disable-next-line no-use-before-define
+  static instance: CallAnalyzerMetrics;
+  meetingCollection: any;
+  webex: any;
+  latencies: CallAnalyzerLatencies;
+
+  /**
+   * Constructor
+   * @constructor
+   * @public
+   */
+  constructor() {
+    if (!CallAnalyzerMetrics.instance) {
+      CallAnalyzerMetrics.instance = this;
+      this.meetingCollection = null;
+      this.latencies = new CallAnalyzerLatencies();
+    }
+
+    // eslint-disable-next-line no-constructor-return
+    return CallAnalyzerMetrics.instance;
```

Comment:
surprisingly this work as it is expected. Although you export a single instance, this is useful to prevent creating more instances via `new instance.constructor`